Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -261,12 +262,19 @@ extension NSManagedObjectContext: MessageDatabaseSession {
         message.isSilent = false
         message.reactionScores = [:]
         message.attachments = try Set(
-            attachments.enumerated().map { index, seed in
+            attachmentSeeds.enumerated().map { index, seed in
                 let id = AttachmentId(cid: cid, messageId: message.id, index: index)
                 let dto = try createNewAttachment(seed: seed, id: id)
                 return dto
             }
+        ).union(
+            attachments.enumerated().map { index, attachment in
+                let id = AttachmentId(cid: cid, messageId: message.id, index: index + attachmentSeeds.count)
+                let dto = try createNewAttachment(attachment: attachment, id: id)
+                return dto
+            }
         )
```

Comment:
This part is very hard to visually parse. Can you please introduce some local variables to make it easier to understant?